### ISSUE TITLE
Accessibility: Fieldsets should always have a legend as their first child element

### ISSUE THREAD

**Issue by philippsauer** [intershop/intershop-pwa#1835]
**Actual Behavior**

Currently, some `<fieldset>` elements within the PWA's forms are being rendered without a corresponding `<legend>` element as their first child. In some cases, `<fieldset>` elements might be present but are not semantically associated with a clear heading via `<legend>`, or the `<legend>` might be missing entirely.  Example: The address form in the "My Account" area (see Screenshot).

A fieldset without a legend is semantically incomplete and loses its primary goal of sematic grouping, especially for accessibility. Without the <legend> label, a screen reader cannot effectively communicate the purpose of the grouped fields. It would hardly differ from a regular <div> element that encloses a group of fields, but without the specific semantic meaning or the accessibility benefit of the <fieldset>.

**Expected Behavior**

Every `<fieldset>` element used for grouping form controls in the Intershop PWA should explicitly include a `<legend>` element as its first child. The `<legend>` element's content should provide a clear and concise description of the grouped controls (e.g., "Personal Details", "Shipping Address", "Company Details").

**Steps to Reproduce the Bug**

1. Go to My Account -> Addresses -> Add Address
2. Observe that the form contains multiple flieldsets, but no legend element.

**Additional Context, like Screenshots, Log File Snippets etc.**

A missing legend element violates the test specified in the following page: https://www.w3.org/WAI/WCAG22/Techniques/html/H71

WCAG criteria are violated:
[1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships) (Sufficient when used for making information and relationships conveyed through presentation programmatically determinable)
[3.3.2: Labels or Instructions](https://www.w3.org/WAI/WCAG22/Understanding/labels-or-instructions) (Sufficient)

![Image](https://github.com/user-attachments/assets/e23f6f55-4416-42bc-bbdf-25c9740c211e)

[AB#107473](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/107473)
